### ISSUE TITLE
using specific port for gitlab server

### ISSUE THREAD

**Issue by SirbitoX** [BigBrassBand/gitlab2tsv#1]
Hi,

I call the following command:
``` java -jar gitlab2tsv.jar http://myip:8001 username password http all > all.tsv```
but I got the following exception from GitLab:
``` Exception in thread "main" java.net.ConnectException: Connection refused ```
Is anything to do for specific port (e.g. 8001) in this command??

Thanks.

**Comment by brainopener** [comment]
Hello,

The connection to GitLab is handled by [this library](https://github.com/timols/java-gitlab-api).

I've looked through that code and it's using the built-in Java HTTP library which supports URLs like _http://myip:8001_.

Can you triple check that _http://myip:8001_ works with another command line tool like curl or wget? If it does, I'll take a harder look. If it doesn't, then I think you have something else going on.

**Comment by SirbitoX** [comment]
@brainopener thanks for response :+1: 

I checked curl and wget and both of them works fine. Although I got  the following response with curl:
``` <html><body>You are being <a href="http://myip:8001/users/sign_in">redirected</a>.</body></html> ```
and as you see, it is because not sending authentication information to GitLab URL.
Anything else to do?

**Comment by brainopener** [comment]
So I just setup a new Gitlab CE instance on CentOS 7. I moved it to port 8001.

Here's what happens:
```
java -jar gitlab2tsv.jar http://192.168.0.19:8001 root password http mine
origin	username	password
http://localhost:8001/root/test-project.git	root	password

```

So... it works.

That make me thinks that the error you're seeing may not be related to the port.

Can you paste in the complete and total exception your seeing along with all of the stack trace?

**Comment by SirbitoX** [comment]
I just tried to execute the command from another server (not the one that GitLab is located on) and it worked fine.
But I have encountered another problem.
When I use the "Bulk Change->Import Configuration" and choose the exported ```tsv``` file, I get the following error:
```Validation : 100%```
```Creating repository http://213.233.175.250:8001/abresalamat/abresalamat-main.git : 100%```
```Internal plugin error. org.eclipse.jgit.api.errors.TransportException http://213.233.175.250:8001/abresalamat/abresalamat-main.git: cannot open git-upload-pack```
```Done with errors```
Do you have any idea?

**Comment by brainopener** [comment]
If you check the JIRA log, there may be a more complete exception -- including a "caused by" with more info.

That "cannot open git-upload-pack" error is coming out of the jgit library. It could be a variety of things -- incomplete access to the repo, etc...

If you [generate a JIRA support .ZIP](https://bigbrassband.freshdesk.com/support/solutions/articles/1000067480-how-to-create-the-support-zip-file-) and e-mail support, they can figure it out.

Alternatively, look for the error in the JIRA log and paste in the full error here.

**Comment by SirbitoX** [comment]
`Error adding repository http://myip:8001/projects/myproject.git
com.xiplink.jira.git.exception.OperationException: Internal plugin error. org.eclipse.jgit.api.errors.TransportException http://myip:8001/projects/myproject.git: cannot open git-upload-pack
        at com.xiplink.jira.git.MultipleGitRepositoryManagerImpl.setupRepository(MultipleGitRepositoryManagerImpl.java:859)
        at com.xiplink.jira.git.rest.ConfigurationResource$2.run(ConfigurationResource.java:344)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.eclipse.jgit.api.errors.TransportException: http://myip:8001/projects/myproject.git: cannot open git-upload-pack
        at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:139)
        at org.eclipse.jgit.api.CloneCommand.fetch(CloneCommand.java:193)
        at org.eclipse.jgit.api.CloneCommand.call(CloneCommand.java:133)
        at com.xiplink.jira.git.MultipleGitRepositoryManagerImpl.runCloneCommand(MultipleGitRepositoryManagerImpl.java:740)
        at com.xiplink.jira.git.MultipleGitRepositoryManagerImpl.setupRepository(MultipleGitRepositoryManagerImpl.java:855)
        ... 2 more`
`Caused by: org.eclipse.jgit.errors.TransportException: http://myip:8001/projects/myproject.git: cannot open git-upload-pack
        at org.eclipse.jgit.transport.TransportHttp.connect(TransportHttp.java:518)
        at org.eclipse.jgit.transport.TransportHttp.openFetch(TransportHttp.java:296)
        at org.eclipse.jgit.transport.FetchProcess.executeImp(FetchProcess.java:136)
        at org.eclipse.jgit.transport.FetchProcess.execute(FetchProcess.java:122)
        at org.eclipse.jgit.transport.Transport.fetch(Transport.java:1138)
        ... 6 more`
`Caused by: java.net.ConnectException: Connection refused myip
        at org.eclipse.jgit.util.HttpSupport.response(HttpSupport.java:181)
        at org.eclipse.jgit.transport.TransportHttp.connect(TransportHttp.java:469)
        ... 11 more
`

The problem comes from where the GitLab (port 8001) and JIRA (port 8002) server is the same. The last exception of stack trace shows it: ```Connection refused myip```
So I tested it with ```curl myip:8001``` and I got ```curl: (7) couldn't connect to host```. But I wonder when I execute the mentioned command on another server I get the correct response.

**Comment by brainopener** [comment]
If I were there, the next thing I would do to diagnose is this:

- Connect to the JIRA server over ssh
- Make sure that git is installed
- Try to clone the git repo: git clone http://myip:8001/projects/myproject.git

If that fails, then there's no hope of the Git JIRA plugin cloning it either. You'll need to then diagnose why the regular Git client can successfully clone.

**Comment by SirbitoX** [comment]
I followed your steps and I got:
`Initialized empty Git repository in /mypath/myproject/.git/
error: Failed connect to myip:8001; Operation now in progress while accessing http://myip:8001/projects/myproject.git/info/refs`
`fatal: HTTP request failed`.
The problem is the one I mentioned before. For some reason ```myip:8001``` address will not resolve on its own server. I searched a lot but I got nothing.
@brainopener Thank you anyway :+1:
